### ISSUE TITLE
Default value for flag if not passed

### ISSUE THREAD

**Issue by Leon0402** [CLIUtils/CLI11#581]
Hi, 

it would be convient if the library would set a value if a flag is **not** passed. I did search a little bit in the documentation and it seemed like this should be possible: 

``` 
bool flag;
command->add_flag("--flag", flag, "Help string")
        ->default_val(false);
```
So flag is set to True if --flag is passed, but not to flag if it's not passed. At least not by the library, as a matter of fact, the default value is false here and I could do something like: 

``` 
bool flag = false;
command->add_flag("--flag", flag, "Help string")
        ->default_val(false);
```

But I parse multiple times, so I can't rely on a default value here. The library needs to set the flag. Is that somehow possible or could be supported? 
Using the existing `default_val` seems logical to me for this.

**Comment by phlptp** [comment]
I think you want the `run_callback_for_default`  option set for the flag.  This will execute the option callback (which by default is setting the variable) for all cases.  Which if nothing is passed will use the `default_val`  This can't be enabled by default since that would cause a lot of unexpected behavior so it needs to be enabled for the options desired.

**Comment by phlptp** [comment]
this is similar to #628 and the docs addressed in #629

**Comment by phlptp** [comment]
as of the 2.1 release there is a `force_callback` option to trigger the callback regardless of whether the option is passed.  if no option is passed it will use the default argument.  

This should address this issue in a much more direct way, than previously explained.
